### PR TITLE
USB plugged priority

### PR DESCRIPTION
**Pull request by simonegu**
**Problem**
If the bootloader UART is also used in the application for telemetry it happens that other devices may write when you are rebooting. Since when a USB cable is plugged in it will extend the Bootloader timeout, this will result in having the board stuck in the bootloader mode because it received some bytes on the UART port.

Adding `BOOT_DEVICES_FILTER_ONUSB USB0_DEV` in your board config will just take inputs from USB when a USB cable is plugged in.

Unfortunately I don't have all the boards to test out. Help there would be appreciated.

### CODE REVIEW

**Comment by ksschwabe** [comment]
@simonegu: You could also just set the INTERFACE_USART to 0, and then it would not use the UART at all to bootload. Would this not be a simpler solution to your problem? Or do you still want to be able to bootload via the UART?

File: bl.c [hunk 1]
```diff
@@ -388,12 +388,13 @@ static int
...
 {
 	int c = -1;
+	char devices = board_get_devices();
 
 	/* start the timeout */
 	timer[TIMER_CIN] = timeout;
 
 	do {
-		c = cin();
+		c = cin(devices);
```
Comment (davids5) [inline bl.c]: use `cin(board_get_devices())` and eliminate Line 391

File: bl.h [hunk 2]
```diff
@@ -120,5 +121,5 @@ extern int check_silicon(void);
 
 extern void cinit(void *config, uint8_t interface);
 extern void cfini(void);
-extern int cin(void);
+extern int cin(char devices);
```
Comment (davids5) [inline bl.h]: Chage type

**Comment by davids5** [comment]
@ksschwabe - this adds  run time checking for USB connected and run time disable the serial. So that setting a board to have BOOT_DEVICES_FILTER_ONUSB = USB0_DEV will disable the serial ports as input sources only when USB is connected.

**Comment by simonegu** [comment]
@ksschwabe yes, the main idea is to have both but if a USB cable is plugged in just accept from the USB interface.

**Comment by simonegu** [comment]
@davids5 Yes :smile:

**Pull request merged**